sites/cheerpj/src/content/docs/11-guides/implementing-native-methods.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Implementing Native Methods
+description: Java Native Interface (JNI) with CheerpJ
+---
+
+CheerpJ allows you to implement Java native methods (typically written in C/C++ or another AOT-compiled language) directly in JavaScript, similar to the Java Native Interface (JNI) in standard Java.
+
+In Java, native methods are identified by the `native` keyword in their declaration. These methods are not implemented in Java but are instead defined in an external language, which in the case of CheerpJ, is JavaScript.
+
```

Comment:
We could already add the steps that we will explain here, something like "Adding natives to CheerpJ only requires a few steps: 1. Define a native function in Java 2. Implement it in Java Script 3. Pass it to CheerpJ". You could also a bit of context why natives would be useful.